src/foam/core/Currency.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.core',
+  name: 'Currency',
+  extends: 'foam.core.Unit',
+
+  documentation: `The base model for storing, using and managing currency information.
+    All class properties require a return of *true* in order to pass.`,
+
+  ids: [
+    'alphabeticCode'
+  ],
+
+  javaImports: [
+    'foam.util.SafetyUtil'
+  ],
+
+  tableColumns: [
+    'name',
+    'alphabeticCode',
+    'country',
+    'symbol',
+    'emoji'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'name',
+      documentation: `This is the [ISO 4217](https://www.iso.org/iso-4217-currency-codes.html)
+        international standard for currency codes.`,
+      required: true
+    },
+    {
+      class: 'String',
+      name: 'alphabeticCode',
+      label: 'Code',
+      documentation: 'The alphabetic code associated with a type of currency.',
+      required: true
+    },
+    {
+      class: 'Long',
+      name: 'numericCode',
+      documentation: 'The numeric code associated with a type of currency.',
+      required: true
+    },
+    {
+      class: 'Int',
+      name: 'precision',
```

Comment:
Why repeat these properties (name, alphabeticCode, precision) that are already in the base class?

Reply:
thanks Kate!